tuf/ngclient/_internal/requests_fetcher.py

```diff
@@ -86,58 +87,62 @@ def fetch(self, url, required_length):
-                    # Furthermore, we do not wish to download an extremely
-                    # large file in one shot. Before beginning the round, sleep
-                    # (if set) for a short amount of time so that the CPU is not
-                    # hogged in the while loop.
-                    if settings.SLEEP_BEFORE_ROUND:
-                        time.sleep(settings.SLEEP_BEFORE_ROUND)
-
-                    read_amount = min(
-                        settings.CHUNK_SIZE,
-                        required_length - bytes_received,
-                    )
-
-                    # NOTE: This may not handle some servers adding a
-                    # Content-Encoding header, which may cause urllib3 to
-                    #  misbehave:
-                    # https://github.com/pypa/pip/blob/404838abcca467648180b358598c597b74d568c9/src/pip/_internal/download.py#L547-L582
-                    data = response.raw.read(read_amount)
-                    bytes_received += len(data)
-
-                    # We might have no more data to read. Check number of bytes
-                    # downloaded.
-                    if not data:
-                        logger.debug(
-                            "Downloaded %d out of %d bytes",
-                            bytes_received,
-                            required_length,
-                        )
-
-                        # Finally, we signal that the download is complete.
-                        break
-
-                    yield data
-
-                    if bytes_received >= required_length:
-                        break
-
-            except urllib3.exceptions.ReadTimeoutError as e:
-                raise exceptions.SlowRetrievalError(str(e))
-
-            finally:
-                response.close()
-
-        return chunks()
+        return self._chunks(response, required_length)
+
+    @staticmethod
+    def _chunks(
+        response: "requests.Response", required_length: int
+    ) -> Iterator[bytes]:
+        """A generator function to be returned by fetch. This way the
+        caller of fetch can differentiate between connection and actual data
+        download."""
+
+        try:
+            bytes_received = 0
+            while True:
+                # We download a fixed chunk of data in every round. This is
+                # so that we can defend against slow retrieval attacks.
+                # Furthermore, we do not wish to download an extremely
+                # large file in one shot. Before beginning the round, sleep
+                # (if set) for a short amount of time so that the CPU is not
+                # hogged in the while loop.
+                if settings.SLEEP_BEFORE_ROUND:
+                    time.sleep(settings.SLEEP_BEFORE_ROUND)
+
+                read_amount = min(
+                    settings.CHUNK_SIZE,
+                    required_length - bytes_received,
+                )
```

Comment:
let's get rid of settings: 
* CHUNK_SIZE: this is a RequestsFetcher internal detail. make it an instance attribute (maybe with a smaller default than the current huge number)?
* SLEEP_BEFORE_ROUND: if this truly is an issue we should fix the issue -- there is no reason for us to be running a hot while loop for downloading things, that's just crazy.  #338 documents that that is indeed how the code used to work... but that was a long time ago, and the code looked entirely different. I'd say let's spend a bit of time making sure the loop is sane (and doesn't hog cpu) and let's remove SLEEP_BEFORE_ROUND

Reply:
for reference, pip and requests use default chunk size of 1024*10

Reply:
this is alread being handled 1470 so maybe forget the comment here -- we should get rid of sleep before round but doesn't need to happen here.

Reply:
I opened https://github.com/theupdateframework/tuf/issues/1484 for it and we can continue discussing settings in the #1470 thread.